core/src/main/java/org/jruby/exceptions/RaiseException.java

```diff
@@ -128,7 +128,19 @@ private void preRaise(ThreadContext context, IRubyObject backtrace, boolean capt
     }
 
     private void setStackTraceFromException() {
-        setStackTrace(javaTraceFromRubyTrace(exception.getBacktraceElements()));
+        RubyStackTraceElement[] rubyTrace = exception.getBacktraceElements();
+        if (rubyTrace.length > 5 && "getStackTrace".equals(rubyTrace[0].getMethodName())) { // -Xbacktrace.style=raw
+            int skip = 0;
+            if ("preRaise".equals(rubyTrace[4].getMethodName())) {
+                skip = 5;
+            } else if ("preRaise".equals(rubyTrace[3].getMethodName())) {
+                skip = 4;
+            }
+            // NOTE: we could skip more useless Throwable.<init> hierarchy constructor trace
+            //  up to org.jruby.exceptions.RaiseException.from ?
+            rubyTrace = Arrays.copyOfRange(rubyTrace, skip, rubyTrace.length);
+        }
+        setStackTrace(javaTraceFromRubyTrace(rubyTrace));
     }
```

Comment:
I wonder if we can just filter out all frames where the package starts with "java." might be safe too? All traces we initiate should be rooted as close to the Ruby code or core class code as possible.

Reply:
in that case I think we would reach JRuby's `JumpException` constructor as top.
was wondering whether to go further - in relation to the Throwable ctor hierarchy.
we might want to skip all of those ctors so a "java." prefix won't do ... or maybe we do not :yin_yang:  
intentionally did not push this too far and rather check at exact places (which is quite ugly but working fine), for now to avoid more costs in looping the whole stack-trace for another time.

Reply:
Yeah I'm fine pushing any further filtering to a later discussion.